src/java/org/apache/cassandra/batchlog/Batch.java

```diff
@@ -113,6 +127,22 @@ public void serialize(Batch batch, DataOutputPlus out, int version) throws IOExc
             }
         }
 
+        @VisibleForTesting
+        public void reserialize(Batch batch, DataOutputPlus out, int version) throws IOException
```

Comment:
I still think it would we useful to have a comment about the purpose of the method, probably very similar to the comment in the `Instance#serializeMessage` method where `reserialize` is called.

Reply:
Agreed. If we keep this method, rather than exposing the encoded mutations, I'll add it...